src/main/java/me/zeroeightsix/kami/gui/hudgui/component/HudButton.kt

```diff
@@ -4,8 +4,9 @@ import me.zeroeightsix.kami.gui.hudgui.HudElement
 import me.zeroeightsix.kami.gui.hudgui.KamiHudGui
 import me.zeroeightsix.kami.gui.rgui.component.BooleanSlider
 import me.zeroeightsix.kami.util.math.Vec2f
+import me.zeroeightsix.kami.util.translation.TranslationKeySetText
 
-class HudButton(val hudElement: HudElement) : BooleanSlider(hudElement.name, 0.0, hudElement.description) {
+class HudButton(val hudElement: HudElement) : BooleanSlider(TranslationKeySetText(hudElement.name.value), 0.0, hudElement.description) {
```

Comment:
The naming is really terrible for this. I don't think we import Il8n anywhere, we could just call this
```
Il8n("HudButton")
```

Reply:
The issue with this is that it changes its name depending on what variables are fed in, if we did 

> 
> ```
> Il8n("HudButton")
> ```

then it would have to always be named the same thing. If you can think of a better way to do this then please say, however I don't believe that this solution would work.

Reply:
I'm not sure what you mean by that, can you explain differently?

Reply:
Sorry, I have just looked at this again, my method of doing this could just be hudelement.name, however this changes depending on the Hud element being displayed,  this means that you can't hard code the name of the element. I have probably explained this really badly, if so please just tell me and I will try again tomorrow morning.